src/web/components/ParticipantManagement/UpdateParticipantDialog.tsx

```diff
@@ -3,49 +3,79 @@ import { SubmitHandler } from 'react-hook-form';
 
 import { ApiRoleDTO } from '../../../api/entities/ApiRole';
 import { ParticipantDTO } from '../../../api/entities/Participant';
+import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
 import { UpdateParticipantForm } from '../../services/participant';
 import { sortApiRoles } from '../../utils/apiRoles';
 import { Dialog } from '../Core/Dialog';
 import { Form } from '../Core/Form';
 import { MultiCheckboxInput } from '../Input/MultiCheckboxInput';
+import { TextInput } from '../Input/TextInput';
 
-type UpdateParticipantDialogProps = {
+type UpdateParticipantDialogProps = Readonly<{
   triggerButton: JSX.Element;
   participant: ParticipantDTO;
   onUpdateParticipant: (form: UpdateParticipantForm, participant: ParticipantDTO) => Promise<void>;
   apiRoles: ApiRoleDTO[];
-};
+  participantTypes: ParticipantTypeDTO[];
+}>;
 
 function UpdateParticipantDialog({
   triggerButton,
   participant,
   onUpdateParticipant,
   apiRoles,
+  participantTypes,
 }: UpdateParticipantDialogProps) {
   const [open, setOpen] = useState(false);
+  const [typesAndRolesError, setTypesAndRolesError] = useState(false);
 
   const onSubmit: SubmitHandler<UpdateParticipantForm> = async (formData) => {
-    await onUpdateParticipant(formData, participant);
-    setOpen(false);
+    if (formData.participantTypes.length === 0 || formData.apiRoles.length === 0) {
+      setTypesAndRolesError(true);
+    } else {
```

Comment:
Per Kimberly, we should not be able to update a participant that removes all participant types or remove all API roles.  Participants must have at least one of each.